### ISSUE TITLE
asBoolean() and convert('boolean') behave differently on an integer

### ISSUE THREAD

**Issue by andreyvk** [orientechnologies/orientdb#7284]
### OS: Ubuntu 14.04

Hi,

I've come across something that I'm not sure if it was done intentionally. Consider the following test case:

```sql
create class Test
insert into Test content {"list": ["a", "b"]}
```

Then the following queries give different results:
```sql
select list.size().asBoolean() as hasList from Test
select list.size().convert('boolean') as hasList from Test
```

The first query: Studio says that 1 record is returned, but nothing is shown (**unexpected result**)
The second query: returns one record with `hasList` attribute set to `true` (**expected result**)

Please, see if this is a problem or not.

**Comment by maggiolo00** [comment]
hi @andreyvk 

thanks for this issue.
yes seems that the second one is correct and the first one has a bug.
I will provide a fix but for 2.2.x. Do you need this in 2.1.x?

**Comment by andreyvk** [comment]
@maggiolo00 Thanks! No fix is needed. I'll just go with `convert` option

**Comment by maggiolo00** [comment]
Thanks @andreyvk

**Comment by maggiolo00** [comment]
Fixed in 2.2.x/develop
